src/cargo/ops/cargo_output_metadata.rs

```diff
@@ -39,10 +39,47 @@ pub fn output_metadata<'a>(opt: OutputMetadataOptions,
 #[derive(RustcEncodable)]
 pub struct ExportInfo {
     packages: Vec<Package>,
-    resolve: Resolve,
+    resolve: MetadataResolve,
     version: u32,
 }
 
+struct MetadataResolve(Resolve);
```

Comment:
There is an option of making this implementation the default, and moving current implementation inside `lockfile` mod. It's not that simple though because of privacy issues.